doc/design/parallel-node-config.md

```diff
@@ -0,0 +1,250 @@
+	Interfaces    InterfaceExts `json:"interfaces,omitempty"`
+    // +kubebuilder:validation:Enum=Idle;Draining,Draining_MCP_Paused,Succeeded,Failed,InProgress
+	DrainStatus    string        `json:"drainStatus,omitempty"`
+	SyncStatus    string        `json:"syncStatus,omitempty"`
+	LastSyncError string        `json:"lastSyncError,omitempty"`
+}
+```
+
+#### Extend existing CR SriovNetworkPoolConfig
+SriovNetworkPoolConfig is used only for OpenShift to provide configuration for
+OVS Hardware Offloading. We can extend it to add configuration for the drain
+pool. E.g.:
+
+```golang
+// DrainConfigSpec defines node pool drain configuration
+type DrainConfigSpec struct {
+    // Number of nodes can be configured in parallel
+    // 0 means no limit, all nodes will be configured in parallel
+    // +kubebuilder:default:=1
+    // +kubebuilder:validation:Minimum:=0
+    MaxParallelNodeConfiguration int `json:"maxParallelNodeConfiguration,omitempty"`
+}
+
+// SriovNetworkPoolConfigSpec defines the desired state of SriovNetworkPoolConfig
+type SriovNetworkPoolConfigSpec struct {
+    // +kubebuilder:validation:Minimum=0
+    // +kubebuilder:validation:Maximum=99
+    // Priority of the SriovNetworkPoolConfig, higher priority policies can override lower ones.
+    Priority int `json:"priority,omitempty"`
+    // OvsHardwareOffloadConfig describes the OVS HWOL configuration for selected Nodes
+    OvsHardwareOffloadConfig OvsHardwareOffloadConfig `json:"ovsHardwareOffloadConfig,omitempty"`
+    // NodeSelectorTerms is a list of node selectors to apply SriovNetworkPoolConfig
+    NodeSelectorTerms *v1.NodeSelectorTerm `json:"nodeSelectorTerms,omitempty"`
+    DrainConfig DrainConfigSpec `json:"drainConfig,omitempty"`
+}
+```
+
+```yaml
+apiVersion: v1
+kind: SriovNetworkPoolConfig
+metadata:
+name: pool-1
+namespace: network-operator
+spec:
+priority:  44
+drainConfig:
+  maxParallelNodeConfiguration: 5
+nodeSelectorTerms:
+  - matchExpressions:
+    - key: some-label
+      operator: In
+      values:
+        - val-2
+  - matchExpressions:
+    - key: other-label
+      operator: "Exists"
+      DrainConfig:
+      maxUnavailable: 5
+      ovsHardwareOffloadConfig: {}
+```
+
+Default pool should be always created with the lowest priority:
+```yaml
+apiVersion: v1
+kind: SriovNetworkPoolConfig
+metadata:
+name: default
+namespace: network-operator
+spec:
+priority: 0
+DrainConfig:
+maxUnavailable: 1
+```
+
+
+### Implementation Details/Notes/Constraints
+The implementation of this feature is very complex and requires a lot of changes to different parts of SR-IOV Network
+Operator. To not introduce breaking changes we have to split this effort to several phases:
+* implement new Drain controller without user-facing API changes:
+  it will proceed only one node configuration at the same time and doesn't require API changes
+* drop `SriovNetworkNodeState` usage and move this annotation values into the `SriovNetworkNodeState` object
```

Comment:
I think here is to drop the annotation from the node object

Reply:
fixed